### DISCUSSION TITLE
What happens if getElementAttr is called with an invalid ID

### DISCUSSION THREAD

**Discussion by andre2007** [rism-digital/verovio#3894] [Q&A]
Hi,
I trying to track down a very weird behavior in my application. For some reasons `renderToTimemap` returns in `on` array a xml ID which causes weired memory issues when calling method `getElementAttr`.

It looks like that calling method `getElementAttr` with an invalid xml ID causes strange behavior. What is the expected behavior if the method is called with an invalid XML ID? Does it make sense to return a JSON like this from the method `getElementAttr`, in case the xml Id is invalid?

```json
{
"error": true,
"errorText": "XML Id ABC is invalid"
}
````

Kind regards
André

**Comment by ahankinson** [comment]
Please share a file that causes the problem.

**Comment by lpugin** [comment]
Moved to the issue tracker #3895

**Reply by lpugin** [reply]
To your question precisely: it returns an empty JSON object.

**Reply by lpugin** [reply]
See the sentence added here https://book.verovio.org/toolkit-reference/toolkit-methods.html#getelementattr
